java/arcs/sdk/wasm/WasmMultipleHandles.kt

```diff
@@ -0,0 +1,781 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.sdk.wasm
+
+/** Combined Handle to allow events on two handles to trigger actions. */
+class WasmTupleHandle<T, U>(
+    val handle1: WasmHandleEvents<T>,
+    val handle2: WasmHandleEvents<U>
+) {
+
+    /**
+     * Trigger a callback when either of the handles updates. The callback will receive the latest
+     * entities from both handles.
+     * */
+    fun onUpdate(action: (T?, U?) -> Unit) {
+        handle1.onUpdate { e ->
+            action(e, handle2.getContent())
+        }
+
+        handle2.onUpdate { e ->
+            action(handle1.getContent(), e)
+        }
+    }
+}
+
+/** Combined Handle to allow events on three handles to trigger actions. */
+class WasmTripleHandle<T, U, V>(
+    val handle1: WasmHandleEvents<T>,
+    val handle2: WasmHandleEvents<U>,
+    val handle3: WasmHandleEvents<V>
+) {
+
+    /**
+     * Trigger a callback when any of the handles updates. The callback will receive the latest
+     * entities from all handles.
+     * */
+    fun onUpdate(action: (T?, U?, V?) -> Unit) {
+        handle1.onUpdate { e ->
```

Comment:
Why is this duplication needed, can this just be:
listOf(handle1, handle2, handle3).forEach { it.onUpdate { _ -> action(handle1.getContent(), handle2.getContent(), handle3.getContent()) } }
?

Reply:
On a similar note: Both classes and functions can have vararg inputs. If each of the generic types (`T`, `U`, `V`, etc.) were covariant instead of totally free (i.e. `T extends WasmEntity`), then only one generic would be needed. Thus, it seems within reach to avoid creating event handle classes and functions.

Reply:
Unfortunately this doesn't work as collections give back a Set<WasmEntity>.